lib/pipeline.js

```diff
@@ -291,43 +332,69 @@ class PipelineModel extends BaseModel {
                     .then((existingJobs) => {
                         const jobsToSync = [];
                         const jobsProcessed = [];
+                        const triggersToSync = [];
                         const parsedConfigJobNames = Object.keys(parsedConfig.jobs);
+                        const pipelineId = this.id;
+                        let requiresList;
```

Comment:
Can you set `requiresList = []` here? Then you don't need to have line 354 and the default value on line 363

Reply:
Sure. I can remove 354 with this but still need `[]` as default value for the case where it sets 
`requireList = permutations[0].requires` and it's set to `undefined` if requires is not there.

Reply:
Added the default case inside the other if instead :)